files/ebs_volume.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+until ping -c 1 google.com &> /dev/null; do
+  echo "waiting for outbound connectivity"
+  sleep 5
+done
+
+# Set common variables used throughout the script.
+imds_token=$( curl -Ss -H "X-aws-ec2-metadata-token-ttl-seconds: 300" -XPUT 169.254.169.254/latest/api/token )
+instance_id=$( curl -Ss -H "X-aws-ec2-metadata-token: $imds_token" 169.254.169.254/latest/meta-data/instance-id )
+availability_zone=$( curl -Ss -H "X-aws-ec2-metadata-token: $imds_token" 169.254.169.254/latest/meta-data/placement/availability-zone )
+volume_id=""
+
+# Search for an available EBS volume to attach to the instance. Wait one minute for a volume to become available,
+# if no volume is found - create new one, attach, format and mount the volume.
+
+for i in $(seq 1 6); do
+
+  volume_id=$(
+    aws --cli-connect-timeout 300 ec2 describe-volumes \
+      --filters "Name=status,Values=available" "Name=availability-zone,Values=$availability_zone" "Name=tag:Name,Values=${name}-graphdb-data" \
+      --query "Volumes[*].{ID:VolumeId}" \
+      --output text | \
+      sed '/^$/d'
+  )
+  
+  if [ -z "${volume_id:-}" ]; then
+    echo 'ebs volume not yet available'
+    sleep 10
+  else
+    break
+  fi
+done
+
+if [ -z "${volume_id:-}" ]; then
+
+throughput_option=""
+if [ "${ebs_volume_type}" = "gp3" ] || [ "${ebs_volume_type}" = "gp2" ]; then
```

Comment:
I have added additional checks; now if the volume type is something different than GP2, GP3, IO1, or IO2, it will break the script with the output:
"The volume type is not supported."